packages/devtools_app/lib/src/screens/performance/performance_controller.dart

```diff
@@ -100,14 +106,57 @@ class PerformanceController extends DisposableController
             final frame = FlutterFrame.parse(event.extensionData!.data);
             enhanceTracingController.assignStateForFrame(frame);
             flutterFramesController.addFrame(frame);
-          } else if (event.extensionKind == 'Flutter.RebuiltWidgets') {
-            rebuildCountModel.processRebuildEvent(event.extensionData!.data);
+          } else if (event.extensionKind == 'Flutter.RebuiltWidgets' &&
+              FeatureFlags.widgetRebuildstats) {
+            final data = this.data!;
+            if (_currentRebuildWidgetsIsolate != event.isolate) {
+              data.rebuildCountModel.clearFromRestart();
+            }
+            _currentRebuildWidgetsIsolate = event.isolate;
+            // TODO(jacobr): need to make sure we don't get events from before
+            // the last hot restart. Their data would be bogus.
+            data.rebuildCountModel
+                .processRebuildEvent(event.extensionData!.data);
+            if (data.rebuildCountModel.locationMap.locationsResolved.value ==
+                    false &&
+                !_fetchMissingLocationsStarted) {
+              _fetchMissingRebuildLocations();
+            }
           }
         }),
       );
     }
   }
 
+  void _fetchMissingRebuildLocations() async {
```

Comment:
can this be moved into the rebuild model? After the refactor I did recently, we are trying to keep code related to a single feature in that feature's controller. Maybe there is a use for a WidgetRebuildsController?

Reply:
Sending a quick followup CL to move.